<?xml version="1.0"?>
<launch>
    <!-- config file path  -->
    <arg name="rviz_settings" default="$(find amr_system)/config/rviz/global_path_planner.rviz"/>
    <arg name="map_settings" default="$(find amr_system)/map/map.yaml"/>
    <arg name="obstacle_expander_settings" default="$(find amr_system)/config/param/obstacle_expander.yaml"/>
    <arg name="global_path_planner_settings" default="$(find amr_system)/config/param/global_path_planner.yaml"/>

    <!-- node launch -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(arg rviz_settings)"/>
    <node pkg="amr_system" type="obstacle_expander_node" name="obstacle_expander" output="screen">
        <rosparam command="load" file="$(arg obstacle_expander_settings)"/>
    </node>
    <node pkg="amr_system" type="global_path_planner_node" name="global_path_planner" output="screen">
        <rosparam command="load" file="$(arg global_path_planner_settings)"/>
    </node>
    <node pkg="map_server" type="map_server" name="map_server" args="$(arg map_settings)" output="screen"/>
</launch>
